### PR TITLE
more flexibility when rendering tick labels with tickFormat

### PR DESCRIPTION
**Pull request by bodyflex**
right now the value rendered from `tickFormat` will always be wrapped in a `<text>` node which is quite limiting if you want to do more flexible stuff. this change checks if the value that's returned from `tickFormat` is a React element and renders it instead of the `<text>` in case it is one. the custom element also gets passed the width of the tick container and the total amount of ticks as props in case it wants to do some responsive stuff.

~as such, this would be a breaking change, but it could also be pretty easily modified to wrap those SVG elements with a `<text>` that can be descendants of `<text>` per spec.~ 

if you think this change is OK, i'll write some tests for it and then hopefully it could be merged :)

### CODE REVIEW

**Comment by bodyflex** [comment]
i tried to find existing tests regarding `tickFormat` but i guess there are none yet. should i try to write some?

**Comment by bodyflex** [comment]
@mcnuttandrew added some tests now!

File: docs/axes.md [hunk 1]
```diff
@@ -87,6 +87,28 @@ function mySIPrefixFormatter(value, index, scale, tickTotal) {
...
+    >
+      {props.children}
+    </div>
+  </foreignObject>
+);
+
+function myFormatter(value) {
+  return <MyLabel>{value}</MyLabel>;
+}
+```
```
Comment (mcnuttandrew) [inline docs/axes.md]: Maybe also include an example of using this with the axes?

File: showcase/axes/custom-axis-tick-element.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  }
+
+  render() {
+    return (
+      <XYPlot
+        width={300}
+        height={300}
+        xType="linear"
+        xDomain={[0, 4]}
+        yType="linear"
```
Comment (mcnuttandrew) [inline showcase/axes/custom-axis-tick-element.js]: nit: unnecessary to list xType and yType

File: tests/components/axis-tick-format-tests.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  );
+  t.equal(
+    ticks.at(4).find('text').text(),
+    'Label',
+    'renders text inside text node'
+  );
+  t.end();
+});
+
+test('Axis Format: passes props to custom element', t => {
```
Comment (mcnuttandrew) [inline tests/components/axis-tick-format-tests.js]: nice test

File: docs/axes.md [hunk 4]
```diff
@@ -87,6 +87,28 @@ function mySIPrefixFormatter(value, index, scale, tickTotal) {
...
+
```
Comment (mcnuttandrew) [inline docs/axes.md]: You can include your new showcase example in here! Just use the magic comment like:

```md
<!-- INJECT:"CustomAxesWithLink" -->
```
But with the appropriate chart name

**Comment by bodyflex** [comment]
all review comments should be fixed now @mcnuttandrew

**Comment by mcnuttandrew** [comment]
Sorry for the long delays! This looks good. stamp stamp

**Comment by bodyflex** [comment]
would be awesome if we got a new release with this and other fixes @mcnuttandrew :)

**Comment by mcnuttandrew** [comment]
@benshope mind cutting a patch?

**Comment by benshope** [comment]
@mcnuttandrew happy to!  Bumped to `1.11.5`

**Comment by mcnuttandrew** [comment]
Woot! Big thank! You continue to be wonderful

**Comment by bodyflex** [comment]
amazing, thanks guys!

**Pull request merged**